gemrb/core/DisplayMessage.cpp

```diff
@@ -188,27 +188,24 @@ void DisplayMessage::DisplayString(String text, const Color &color, Scriptable *
 	}
 }
 
-std::vector<std::string> DisplayMessage::GetAllColors() const
+std::map<int, std::string> DisplayMessage::GetAllColors() const
```

Comment:
Why not the map I proposed? There's not only no need for the full range of integer keys, but more bugs are possible that way.